File: fibonacci/Go/fibo_iter.go
package main

func fibo_iter(number uint64) uint64 {
	if number < 2 {
		return number
	}
	var a uint64 = 1
	var b uint64 = 1
	var value uint64 = 0
	for i := 2; i < int(number); i++ {
		value = a + b
		a = b
		b = value
	}
	return value
}

File: fibonacci/C/fibo_iter.c
#include "fibo_iter.h"

bigint fibo_iter(int number) {
    bigint a = 1;
    bigint b = 1;
    bigint value;
    if (number < 2) 
        return number;
    for (int i = 2; i < number; i++) {
        value = a + b;
        a = b;
        b = value;
    }
    return value;
}